.github/workflows/ci-go.yml

```diff
@@ -54,22 +54,18 @@ jobs:
           node-version: 16
           cache: pnpm
 
-      - name: Install dependencies
-        # corepack 0.10 has a race condition in it that can cause the inital
-        # package manger installs to fail.
-        # Once we update to a version of node that ships with corepack 0.12 or
-        # higher we can remove the second install command.
-        # Windows installs global packages to a directory that has lower priority than
-        # the default node install so we also need to edit $PATH
+      - name: Configure corepack
+        # Forcibly upgrade our available version of corepack.
+        # The bundled version in node 16 has known issues.
+        # Prepends the corepack bin dir so that it is always first.
         shell: bash
         run: |
-          pnpm install;
-          if [ "$RUNNER_OS" == "Windows" ]; then
-            npm install --force --global corepack@latest
-            npm config get prefix >> $GITHUB_PATH
-          else
-            npm install --global corepack@latest
-          fi
+          npm install --force --global corepack@latest
+          npm config get prefix >> $GITHUB_PATH
+          corepack enable
+
+      - name: Install dependencies
+        run: pnpm install
```

Comment:
It would be cool if this was `corepack -- install`. That way a change to `packageManager` wouldn't require another change here

Reply:
We've got a secondary pnpm caching setup above this in the file that is also dependent upon this. This is what happens when we're semi-early adopters.

Reply:
(`corepack` is not (yet?) in the business of delegating to package managers for certain functionality.)